Java: Add Guard Classes for checking OS & unify System Property Access (#8032)
github/codeql: java/ql/src/change-notes/2022-02-14-os-guards.md

---
category: minorAnalysis
---
 * Add new guards `IsWindowsGuard`, `IsSpecificWindowsVariant`, `IsUnixGuard`, and `IsSpecificUnixVariant` to detect OS specific guards.

>  * Add new predicate `getSystemProperty` that gets all expressions that retrieve system properties from a variety of sources (eg. alternative JDK API's, Google Guava, Apache Commons, Apache IO, ect..).

Reviewer: > Apache Commons, Apache IO

The library names here might be a misleading / incomplete because their full names are "Apache Commons Lang" and "Apache Commons IO".

>  * Add new predicate `getSystemProperty` that gets all expressions that retrieve system properties from a variety of sources (eg. alternative JDK API's, Google Guava, Apache Commons, Apache IO, ect..).

Reviewer: ```suggestion
 * Add new predicate `getSystemProperty` that gets all expressions that retrieve system properties from a variety of sources (e.g. alternative JDK API's, Google Guava, Apache Commons, Apache IO, etc.).
```

>  * Add new predicate `getSystemProperty` that gets all expressions that retrieve system properties from a variety of sources (eg. alternative JDK API's, Google Guava, Apache Commons, Apache IO, ect..).

Reviewer 2: ```suggestion
 * Add new predicate `getSystemProperty` that gets all expressions that retrieve system properties from a variety of sources (eg. alternative JDK API's, Google Guava, Apache Commons, Apache IO, etc..).
```

Reviewer: Maybe also the following?
- "eg." -> "e.g."
- "etc.." -> "etc."

 * Update "Local information disclosure in a temporary directory" (`java/local-temp-file-or-directory-information-disclosure`) to remove false-positives when OS is properly used as logical guard.

>  * Update "Local information disclosure in a temporary directory" (`java/local-temp-file-or-directory-information-disclosure`) to use `getSystemProperty` to use new `getSystemProperty` predicate.

Reviewer: Duplicate text:
```suggestion
 * Update "Local information disclosure in a temporary directory" (`java/local-temp-file-or-directory-information-disclosure`) to use new `getSystemProperty` predicate.
```

>  * Update "Local information disclosure in a temporary directory" (`java/local-temp-file-or-directory-information-disclosure`) to use `getSystemProperty` to use new `getSystemProperty` predicate.

Reviewer 2: ```suggestion
 * Update "Local information disclosure in a temporary directory" (`java/local-temp-file-or-directory-information-disclosure`) to use new `getSystemProperty` predicate.
```

---
category: minorAnalysis
---
 * Add new guards `IsWindowsGuard`, `IsSpecificWindowsVariant`, `IsUnixGuard`, and `IsSpecificUnixVariant` to detect OS specific guards.
 * Add new predicate `getSystemProperty` that gets all expressions that retrieve system properties from a variety of sources (eg. alternative JDK API's, Google Guava, Apache Commons, Apache IO, ect..).
 * Update "Local information disclosure in a temporary directory" (`java/local-temp-file-or-directory-information-disclosure`) to remove false-positives when OS is properly used as logical guard.

>  * Update "Local information disclosure in a temporary directory" (`java/local-temp-file-or-directory-information-disclosure`) to use `getSystemProperty` to resolve more 

Reviewer 3: This sentence seems to be incomplete or at least missing a period.

Author: Indeed, it is incomplete. My bad
